typescript/rest-hapi/src/plugins/posts.ts

```diff
@@ -66,12 +66,30 @@ export default usersPlugin
 async function feedHandler(request: Hapi.Request, h: Hapi.ResponseToolkit) {
   const { prisma } = request.server.app
 
+  const { searchString, skip, take, orderBy } = request.query
+
+  const or = searchString ? {
+    OR: [
+      { title: { contains: searchString } },
+      { content: { contains: searchString } },
+    ],
+  } : {}
+
   try {
-    const createdPost = await prisma.post.findMany({
-      where: { published: true },
+    const posts = await prisma.post.findMany({
+      where: {
+        published: true,
+        ...or
+      },
       include: { author: true },
+      take: Number(take) || undefined,
+      skip: Number(skip) || undefined,
+      orderBy: {
+        updatedAt: orderBy || undefined
```

Comment:
are these `|| undefined`s necessary here?

Reply:
Yes. If not added, it defaults to `NaN`.
Here's the error that is thrown if ` || undefined` is omitted: 
```bash
PrismaClientValidationError:
Invalid `prisma.post.findMany()` invocation:

{
  where: {
    published: true
  },
  include: {
    author: true
  },
  take: NaN,
        ~~~
  skip: NaN,
        ~~~
  orderBy: {
    updatedAt: 'desc'
  }
}
```